nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
```

Comment:
- [ ] good job implementing the Nameable class, however, it seems like you either forgot to push or didn't create the remaining requirements listed below as well:

Turn your Person class to Nameable

    Make sure that your Person class inherits from Nameable
    Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

Prepare base Decorator

    Make sure that it inherits from Nameable.
    In the constructor assign a nameable object from params to an instance variable.
    Implement the correct_name method that returns the result of the correct_name method of the @nameable.

Prepare CapitalizeDecorator and TrimmerDecorator

    For the CapitalizeDecorator:
        Create a class that inherits from the base Decorator class.
        Implement a method correct_name that capitalizes the output of @nameable.correct_name.
    For the TrimmerDecorator:
        Create a class that inherits from the base Decorator class.
        Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

Reply:
Hello @mikethreels , It seems like some of my commits disappeared. 😮